src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -0,0 +1,231 @@
+                                        && parameters[0].Type.Equals(WellKnownTypes.StreamingContext(compilation));
+                                    var flag2 = methodSymbol.IsOnDeserializationImplementation(compilation);
+                                    var flag3 = methodSymbol.IsDisposeImplementation(compilation);
+                                    var flag4 = methodSymbol.IsFinalizer();
+
+                                    if (!flag1 && !flag2 && !flag3 && !flag4)
+                                    {
+                                        continue;
+                                    }
+                                    
+                                    var result = new HashSet<IInvocationOperation>();
+                                    FindDangerousMethodInvocationOperation(methodSymbol, compilation, dangerousMethodSymbols, visitedMethodSymbols, result);
+
+                                    foreach (var item in result)
+                                    {
+                                        symbolAnalysisContext.ReportDiagnostic(
+                                            item.CreateDiagnostic(
+                                                Rule,
+                                                classSymbol.MetadataName,
+                                                methodSymbol.MetadataName,
+                                                item.TargetMethod.MetadataName));
+                                    }
+                                }
+                            }
+                        },
+                        SymbolKind.NamedType);
+                });
+
+            /// <summary>
+            /// Traverse every invocation in the method to find all the dangerous function it invokes.
+            /// </summary>
+            /// <param name="methodSymbol">The Symbol of the method which is going to be analyzed</param>
+            /// <param name="compilation">Compilation context</param>
+            /// <param name="dangerousMethodSymbols">Symbols reprensent dangerous methods</param>
+            /// <param name="visitedMethodSymbols">Methods that have been analyzed</param>
+            /// <param name="result">The result set of dangerous invocation operations found in the method</param>
+            void FindDangerousMethodInvocationOperation(
+                IMethodSymbol methodSymbol,
+                Compilation compilation,
+                ImmutableArray<IMethodSymbol> dangerousMethodSymbols,
+                HashSet<IMethodSymbol> visitedMethodSymbols,
+                HashSet<IInvocationOperation> result)
+            {
+                // Every method in the source just needs to be analyzed once.
+                if (!visitedMethodSymbols.Add(methodSymbol))
+                {
+                    return;
+                }
+                
+                var blockOperation = methodSymbol.GetTopmostOperationBlock(compilation);
```

Comment:
This is not the recommended way to write IOperation analyzers, this will lead to duplicate binding/semantic analysis between analyzer and compiler. You should instead register an OperationBlockStartAction within the compilation start action. Your current checks in `RegisterSymbolAction` callback should be done in the OperationBlockStartAction callback - the only duplicate checks might be you will check containing type's attributes for every method of a type, but these should be cheap.